src/repository/CheckForUpdateRepo.ts

```diff
@@ -0,0 +1,50 @@
+import { ExtensionContext, Memento } from "vscode";
+import { ConfigurationTarget } from "../ConfigurationTarget";
+
+type LastUpdated = {
+  prevVersion?: string;
+  lastUpdatedAt?: string;
+};
+
+export interface CheckForUpdateRepo {
+  getLastUpdated(target: ConfigurationTarget): LastUpdated;
+  saveLastUpdated(
+    serverVersion: string,
+    lastUpdatedAt: string,
+    target: ConfigurationTarget
+  ): void;
+}
+
+export class DefaultCheckForUpdateRepo implements CheckForUpdateRepo {
```

Comment:
SO it's not a problem that interface is located in the same file which has imports from `vscode` package as long as it doesn't use them, nice. Maybe I'll rewrite/write some unit tests for test explorer then :D